ExampleMod/Projectiles/ExampleLaser.cs

```diff
@@ -111,21 +119,23 @@ public override void AI()
 			{
 				Vector2 diff = mousePos - player.Center;
 				diff.Normalize();
-				projectile.position = player.Center + diff * _moveDist;
-				projectile.timeLeft = 2;
-				int dir = projectile.position.X > player.position.X ? 1 : -1;
-				player.ChangeDir(dir);
-				player.heldProj = projectile.whoAmI;
-				player.itemTime = 2;
-				player.itemAnimation = 2;
-				player.itemRotation = (float)Math.Atan2(diff.Y * dir, diff.X * dir);
-				#endregion
+                projectile.velocity = diff;
+                projectile.netUpdate = true;
 			}
-
-
-			#region Charging process
-			// Kill the projectile if the player stops channeling
-			if (!player.channel)
+            projectile.position = player.Center + projectile.velocity * MOVE_DISTANCE;

```

Comment:
Yes, it should be take out of the statement. The `player` now is the projectile's owner. The key of my code is the` projectile.netUpdate = true;` inside of `if (projectile.owner == Main.myPlayer)`. Which updates the projectiles velocity(which is the unit vector of player's center to the local mouse position). In this way, the laser can shoot to the correct position.